pkg/api/v1alpha1/versionskew.go

```diff
@@ -0,0 +1,37 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+const supportedMinorVersionIncrement = 1
+
+// ValidateVersionSkew validates Kubernetes version skew between valid non-nil versions.
+func ValidateVersionSkew(oldVersion, newVersion *version.Version) error {
+	newVersionMajor := newVersion.Major()
+	newVersionMinor := newVersion.Minor()
+	oldVersionMajor := oldVersion.Major()
+	oldVersionMinor := oldVersion.Minor()
+
+	majorVersionDifference := math.Abs(float64(newVersionMajor) - float64(oldVersionMajor))
+
+	if majorVersionDifference > 0 {
+		return fmt.Errorf("MAJOR version upgrades are not supported. Major version difference between upgrade version (%d.%d) and server version (%d.%d) is %f", newVersionMajor, newVersionMinor, oldVersionMajor, oldVersionMinor, majorVersionDifference)
```

Comment:
curious why use capital letters for MAJOR?
Same for MINOR

Reply:
basically to emphasize which version is causing the issue. we were initially saying `WARNING`, which I figured instead made sense to display the version type.